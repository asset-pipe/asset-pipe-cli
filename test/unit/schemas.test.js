'use strict';

const Joi = require('joi');
const schemas = require('../../lib/schemas');

test('file schema allows for file with .js extension', () => {
    const result = Joi.validate('index.js', schemas.file);
    expect(result.error).toEqual(null);
});

test('file schema allows for file with .css extension', () => {
    const result = Joi.validate('index.css', schemas.file);
    expect(result.error).toEqual(null);
});

test('file schema does not allow for non .js/.css file extensions', () => {
    const result = Joi.validate('index.json', schemas.file);
    expect(result.error).toMatchSnapshot();
});

test('file schema allows for absolute file paths', () => {
    const result = Joi.validate('/path/to/index.js', schemas.file);
    expect(result.error).toEqual(null);
});
