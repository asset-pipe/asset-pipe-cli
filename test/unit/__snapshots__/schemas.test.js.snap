// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file schema does not allow for non .js/.css/.ts/.tsx file extensions 1`] = `[ValidationError: "entrypoint filename" with value "index.json" fails to match the required pattern: /^[a-zA-Z0-9\\/._-]+\\.(js|css|ts|tsx)$/]`;
